### ISSUE TITLE
error: only v1/graphql and v1alpha1/graphql are supported

### ISSUE THREAD

**Issue by andyrue** [hasura/graphql-engine#2823]
I'm not sure if this is a bug or a configuration problem nor if it's an issue with Hasura or Apollo, but I can't seem to get past this error.  

I'm trying to configure websockets and subscriptions using an authorization header, and I'm pretty sure I've got the Apollo side configured correctly, but I keep getting 
`{"event":{"type":"rejected","detail":{"path":"$","error":"only '/v1/graphql', '/v1alpha1/graphql' are supported on websockets","code":"not-found"}}` in the docker console for hasura beta.6.
Queries work as expected using the JWT in the header, but websockets fail.

Here is my apollo link setup.

    const http = httpLink.create({
      uri: 'http://localhost:8080/v1/graphql',
    });

    const ws = new WebSocketLink({
      uri: `ws://localhost:8080/v1/graphql'`,
      options: {
        reconnect: true,
      }
    });

    const authLink = new ApolloLink((operation, forward) => {
      operation.setContext({
          headers: {
              Authorization: `Bearer ${cookieService.get('auth_token')}` || ''
          }
      });
      return forward(operation);
    });

    const link = split(
      ({ query }) => {
        const definition = getMainDefinition(query);
        return definition.kind === 'OperationDefinition' && definition.operation === 'subscription';
      },
      ws,
      http,
    );

    apollo.create({
      link: ApolloLink.from([
        authLink,
        link
      ]),
      cache: new InMemoryCache()
    });`

**Comment by ecthiender** [comment]
@andyrue the error is thrown by the server when the websocket URL has a different path other than `/v1/graphql` or `/v1alpha1/graphql`. But from your above code snippet the URL looks correct. Not sure why this error would happen. Do you have any proxy before the Hasura server?

**Comment by rikinsk13** [comment]
There seems to be a typo. An extra quote at the end of  `ws://localhost:8080/v1/graphql'`,

**Comment by andyrue** [comment]
> There seems to be a typo. An extra quote at the end of `ws://localhost:8080/v1/graphql'`

Oh my word! Thank you...that was it. I spent more than a day configuring, reconfiguring, configuring again using every bit of info I could find on how to set this up. 

Question, is there a way for Hasura to log the path during this error? I kept wondering if it was somehow getting the wrong path, but the error just says `"path": "$"` 

Thanks again for your help!
